encoder/encoder.go

```diff
@@ -38,22 +41,20 @@ func RegisterType(rType reflect.Type) error {
 	); err != nil {
 		return err
 	}
-	initEncModes()
+	initEncAndDecModes()
 	tagNum++
 	return nil
 }
 
-func init() {
-	initEncModes()
-}
-
 // Marshal returns encoding of param v
 func Marshal(v any) ([]byte, error) {
+	initialiseEncoder.Do(initEncAndDecModes)
```

Comment:
This might not work with https://github.com/NethermindEth/juno/commit/3fc38a92541497f3519ba21c19b9a0d0efbec0cb, we should check